app/celery/letters_pdf_tasks.py

```diff
@@ -133,39 +133,44 @@ def collate_letter_pdfs_to_be_sent():
 
     letters_to_print = get_key_and_size_of_letters_to_be_sent_to_print(print_run_deadline)
 
-    for i, letters in enumerate(group_letters(letters_to_print)):
-        filenames = [letter['Key'] for letter in letters]
-
-        hash = urlsafe_b64encode(sha512(''.join(filenames).encode()).digest())[:20].decode()
-        # eg NOTIFY.2018-12-31.001.Wjrui5nAvObjPd-3GEL-.ZIP
-        dvla_filename = 'NOTIFY.{date}.{num:03}.{hash}.ZIP'.format(
-            date=print_run_deadline.strftime("%Y-%m-%d"),
-            num=i + 1,
-            hash=hash
-        )
+    i = 0
+    for zip_folder, letters_list in letters_to_print.items():
+        for letters in group_letters(letters_list):
+            i += 1
+            filenames = [letter['Key'] for letter in letters]
+
+            hash = urlsafe_b64encode(sha512(''.join(filenames).encode()).digest())[:20].decode()
+            # eg NOTIFY.2018-12-31.001.Wjrui5nAvObjPd-3GEL-.ZIP
+            dvla_filename = 'NOTIFY.{date}.{num:03}.{hash}.ZIP'.format(
+                date=print_run_deadline.strftime("%Y-%m-%d"),
+                num=i,
+                hash=hash
+            )
 
-        current_app.logger.info(
-            'Calling task zip-and-send-letter-pdfs for {} pdfs to upload {} with total size {:,} bytes'.format(
-                len(filenames),
-                dvla_filename,
-                sum(letter['Size'] for letter in letters)
+            current_app.logger.info(
+                'Calling task zip-and-send-letter-pdfs for {} pdfs to upload {} with total size {:,} bytes'.format(
+                    len(filenames),
+                    dvla_filename,
+                    sum(letter['Size'] for letter in letters)
+                )
+            )
+            notify_celery.send_task(
+                name=TaskNames.ZIP_AND_SEND_LETTER_PDFS,
+                kwargs={
+                    'filenames_to_zip': filenames,
+                    'upload_filename': dvla_filename
+                },
+                queue=QueueNames.PROCESS_FTP,
+                compression='zlib'
             )
-        )
-        notify_celery.send_task(
-            name=TaskNames.ZIP_AND_SEND_LETTER_PDFS,
-            kwargs={
-                'filenames_to_zip': filenames,
-                'upload_filename': dvla_filename
-            },
-            queue=QueueNames.PROCESS_FTP,
-            compression='zlib'
-        )
 
 
 def get_key_and_size_of_letters_to_be_sent_to_print(print_run_deadline):
     letters_awaiting_sending = dao_get_letters_to_be_printed(print_run_deadline)
```

Comment:
Would it be easier to get the letters by postage and return a list of letter for each postage type. That way we could start sending zip files to ftp sooner and start with first class.

Reply:
Could we still bundle Europe and Rest of the World together into one zip file if we get the letters by postage? Pete told me this is how DVLA wants them: pdf files named with E or N, but later bundled into one zipfile.

Reply:
Ok, I contacted DVLA and they are fine with 4 categories. I can look into the solution you proposed. So far, I did some other changes, you can find them in the last commit ☺️ .

Reply:
Ok this is done now 👍